Makefile

```diff
@@ -0,0 +1,27 @@
+.PHONY: all
+all: check test build
+
+DOCKER_CMD := docker run --rm -v "$(PWD)":/go/src/github.com/openshift/ci-operator:Z -w /go/src/github.com/openshift/ci-operator golang:1.10
```

Comment:
Generally our makefiles don't make docker calls so that we can run them by default inside of containers (vs requiring everyone to have docker locally).  I.e. I have go on my Mac, don't need docker to run it.

Reply:
That's why it is optional. Everything can be run without docker.

Reply:
What if I inverted it? By default run in host env and when some variable is set, run in container. This way all your workflows would be preserved and as an addition you will have an option to have reporoducible build environment in containers.